dipy/viz/ui.py

```diff
@@ -1884,3 +1905,478 @@ def handle_events(self, actor):
+        obj: :class:`vtkActor`
+            The picked actor
+        file_select_menu: :class:`FileSelectMenu2D`
+
+        """
+        all_file_names = file_select_menu.get_all_file_names()
+
+        if (file_select_menu.n_text_actors +
+                file_select_menu.window) < len(all_file_names):
+            file_select_menu.window += 1
+            file_select_menu.fill_text_actors()
+
+        i_ren.force_render()
+        i_ren.event.abort()  # Stop propagating the event.
+
+    def fill_text_actors(self):
+        """ Fills file/folder names to text actors.
+
+        The list is truncated if the number of file/folder names is greater
+        than the available number of text actors.
+
+        """
+        # Flush all the text actors
+        for text_item in self.text_item_list:
+            text_item.text_actor.message = ""
+            text_item.text_actor.actor.SetVisibility(False)
+
+        all_file_names = self.get_all_file_names()
+
+        clipped_file_names = all_file_names[self.window:self.n_text_actors+self.window]
+
+        # Allot file names as in the above list
+        i = 0
+        for file_name in clipped_file_names:
+            self.text_item_list[i].text_actor.actor.SetVisibility(True)
+            self.text_item_list[i].set_attributes(file_name[0], file_name[1])
+            i += 1
+
+    def get_all_file_names(self):
+        """ Gets file and directory names.
+
+        Returns
+        -------
+        all_file_names: list(string)
+            List of all file and directory names as string.
+
+        """
+        all_file_names = []
+
+        directory_names = self.get_directory_names()
+        for directory_name in directory_names:
+            all_file_names.append((directory_name, "directory"))
+
+        file_names = self.get_file_names()
+        for file_name in file_names:
+            all_file_names.append((file_name, "file"))
+
+        return all_file_names
+
+    def get_directory_names(self):
+        """ Re-allots file names to the text actors.
+
+        Uses FileSelectMenuText2D for selecting files and folders.
+
+        Returns
+        -------
+        directory_names: list(string)
+            List of all directory names as string.
+
+        """
+        # A list of directory names in the current directory
+        directory_names = next(os.walk(self.current_directory))[1]
+        directory_names = [os.path.basename(os.path.abspath(dn)) for dn in directory_names]
+        directory_names = ["../"] + directory_names
+
+        return directory_names
+
+    def get_file_names(self):
+        """ Re-allots file names to the text actors.
+
+        Uses FileSelectMenuText2D for selecting files and folders.
+
+        Returns
+        -------
+        file_names: list(string)
+            List of all file names as string.
+
+        """
+        # A list of file names with extension in the current directory
+        file_names = []
+        for extension in self.extensions:
```

Comment:
This should be tested to make sure it returns only files with the desired extensions. Thinking about it, maybe these methods (that one and `get_directory_names`) should be standalone functions in some utils module and tested separately.

Reply:
Which file in the utils module should this go into? Or do I create another file?

Reply:
Hmm, I thought more about it and we don't have to test the behavior of `glob` :P. It should be fine like this.